trelawney/shap_explainer.py

```diff
@@ -0,0 +1,53 @@
+from typing import List, Optional, Dict
+
+import pandas as pd
+import shap as shap
+import sklearn
+import numpy as np
+import operator
+from keras import models
+from sklearn.ensemble import RandomForestClassifier
+from sklearn.linear_model import LogisticRegression
+from sklearn.tree import DecisionTreeClassifier
+from tqdm import tqdm
+from xgboost import XGBClassifier
+
+from trelawney.base_explainer import BaseExplainer
+
+
+class ShapExplainer(BaseExplainer):
+
+    def __init__(self, class_names: Optional[List[str]] = None, categorical_features: Optional[List[str]] = None, ):
```

Comment:
[NICE TO HAVE] you could have an argument to coerce the explainer to the kernel explainer since it is allways applicable. If we don't want to compare a Tree based and the deep SHAP implementations for instance

Reply:
I'm sorry I didn't understand what you mean: we could use the right explainer for each model if we have it, and otherwise the kernel